keps/sig-network/4762-allow-arbitrary-fqdn-as-pod-hostname/README.md

```diff
@@ -0,0 +1,589 @@
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+-->
+
+#### Alpha
+
+- Use the `HostnameOverride` feature gate to implement this feature.
+- Initial e2e tests completed and enabled.
+- Add documentation for feature gates.
+- Add a detailed table to the docs illustrating the mappings between pod hostnames and DNS records under different configurations.
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Make feature gate to be enabled by default.
+- Update the feature gate documentation.
+
+#### GA
+
+- No issues reported during two releases.
+
+### Upgrade / Downgrade Strategy
+
+API server should be upgraded before Kubelets. Kubelets should be downgraded before the API server.
+
+### Version Skew Strategy
+
+No need to consider version skew, as older versions of kubelet will not be concerned with the `hostnameOverride` field in the podSpec.
+
+## Production Readiness Review Questionnaire
+
+<!--
+
+Production readiness reviews are intended to ensure that features merging into
+Kubernetes are observable, scalable and supportable; can be safely operated in
+production environments, and can be disabled or rolled back in the event they
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/1194-prod-readiness.
+
+The production readiness review questionnaire must be completed and approved
+for the KEP to move to `implementable` status and be included in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+-->
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: HostnameOverride
+  - Components depending on the feature gate: kubelet, kube-apiserver
+- [ ] Other
+  - Describe the mechanism:
+  - Will enabling / disabling the feature require downtime of the control
+    plane?
+  - Will enabling / disabling the feature require downtime or reprovisioning
+    of a node?
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Using the feature gate is the only way to enable/disable this feature.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+
+The feature should continue to work just fine.
+
+###### Are there any tests for feature enablement/disablement?
+
+We will manually test enabling and disabling the feature gate.
```

Comment:
This is not acceptable. We have testing framework for feature gate enablement and disablement and unit or integrations are required to ensure that depending on the feature gate the functionality is or is not working.